source/Keyboard.cpp

```diff
@@ -28,6 +28,41 @@ Keyboard::Keyboard(Server *server, struct wlr_input_device *device) : server(ser
         view->close();
     };
   }};
+  shortcuts["Alt+F2"] = {"Toggle fullscreen", [server](){
+    if (wl_list_length(&server->views) >= 1)
+      {
+        View *view = wl_container_of(server->views.next, view, link);
+	if (!view->fullscreen)
+	  {
+	    wlr_xdg_surface_v6_get_geometry(view->xdg_surface, &view->saved);
+	    view->saved.x = view->x;
+	    view->saved.y = view->y;
+	    struct wlr_box *outputBox = wlr_output_layout_get_box(view->server->output->getLayout(), view->getOutput());
+	    wlr_xdg_toplevel_v6_set_size(view->xdg_surface, outputBox->width, outputBox->height);
+	    view->x = 0;
+	    view->y = 0;
+	    wlr_xdg_toplevel_v6_set_fullscreen(view->xdg_surface, true);
+	  }
+	else
+	  {
+	    wlr_xdg_toplevel_v6_set_fullscreen(view->xdg_surface, false);
+	    wlr_xdg_toplevel_v6_set_size(view->xdg_surface, view->saved.width, view->saved.height);
+	    view->x = view->saved.x;
+	    view->y = view->saved.y;
+	  }
+	view->fullscreen = !view->fullscreen;
+      }
+  }};
+  shortcuts["Alt+Tab"] = {"Switch window", [server](){
```

Comment:
What is this?

Reply:
alt tab function to cycle between windows

Reply:
If it works while a window is fullscreen we may have a problem.

Reply:
Actually, in all other system you can "Alt+Tab" while one of your window is in fullscreen

Reply:
In that case we have a problem when a window is already fullscreen and you fulllscreen another window on the same output. Just tested, the results are not good.

Reply:
Still need the "Alt+Tab" feature.